datamarket/components/MyDatasets/DatasetActions.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { DatasetWithIndex } from "../../constants";
+import DownloadIcon from "@mui/icons-material/Download";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import EditIcon from "@mui/icons-material/Edit";
+import {
+  decryptAndDownload,
+} from "../../utils/utils";
+import { Button } from "../Button";
+import { useRouter } from "next/router";
+
+export const DatasetActions = ({
+  isOwner,
+  publicIndex,
+  dataset,
+  accounts,
+  showIcons,
+}: {
+    isOwner?: boolean;
+    publicIndex: number | undefined;
+    dataset: DatasetWithIndex;
+    accounts: string[];
+    showIcons: boolean;
```

Comment:
make sure to format your documents. this seems out of wack

Reply:
Done